flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -863,11 +871,6 @@ public CompletableFuture<String> triggerSavepoint(
         return schedulerNG.triggerSavepoint(targetDirectory, cancelJob, formatType);
     }
 
-    @Override
```

Comment:
we shouldn't remove a public method from a public class. it is also necessary here. it can be kept for backward compatibility.

Reply:
The backward compatibility should already been kept. At this class's parent interface `JobMasterGateway`, this method is supported as 

```
    /**
     * Triggers taking a checkpoint of the executed job.
     *
     * @param timeout for the rpc call
     * @return Future which is completed with the checkpoint path once completed
     */
    default CompletableFuture<String> triggerCheckpoint(@RpcTimeout final Time timeout) {
        return triggerCheckpoint(null, timeout).thenApply(CompletedCheckpoint::getExternalPointer);
    }

Reply:
got it. you added a default impl method in the parent interface.